src/Azure.Functions.Cli/Helpers/VersionHelper.cs

```diff
@@ -184,15 +184,37 @@ public string CoreToolsReleaseNumber
                 }
             }
             public CoreToolsRelease ReleaseDetail { get; set; }
+            public string CoreToolsAssemblyZipFile
+            {
+                get
+                {
+                    var downloadLink = ReleaseDetail?.DownloadLink;
+                    if (string.IsNullOrEmpty(ReleaseDetail?.DownloadLink))
+                    {
+                        return string.Empty;
+                    }
+
+                    Uri uri = new UriBuilder(ReleaseDetail?.DownloadLink).Uri;
+
+                    if (uri.Segments.Length < 4)
+                    {
+                        return string.Empty;
+                    }
+
+                    return uri.Segments[3];
```

Comment:
What if the URL has more than 4 segments? We should not make the assumption about the number of segments in the URL here. You could consider switching the logic to something like below

```csharp
var uri = new Uri(downloadLink);
var zipSegment = uri.Segments.FirstOrDefault(segment => segment.EndsWith(".zip", StringComparison.OrdinalIgnoreCase));

```
I am also not a fan of executing code for a property access (get). Ideally, we should have done this calculation in the constructor, so the property get access is only returning already computed value.

Reply:
@kshyju Thanks for the help.  I just addressed those changes( calucating the value in constructor).  could I get another review.